src/trs/rewrite.rs

```diff
@@ -304,11 +300,196 @@ impl TRS {
             Err(SampleError::OptionsExhausted)
         } else {
             let mut trs = self.clone();
-            trs.utrs
-                .remove_clauses(sample_iter(rng, deletable, 1)?[0])?;
+            trs.utrs.remove_clauses(sample_iter(rng, deletable, 1)?[0])?;
             Ok(trs)
         }
     }
+    /// Replaces one subterm with another subterm in a main Term.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # #[macro_use] extern crate polytype;
+    /// # extern crate programinduction;
+    /// # extern crate rand;
+    /// # extern crate term_rewriting;
+    /// # use programinduction::trs::{TRS, Lexicon};
+    /// # use rand::{thread_rng};
+    /// # use term_rewriting::{Context, RuleContext, Signature, parse_rule, parse_term};
+    /// # fn main() {
+    /// let mut sig = Signature::default();
+    /// let term = parse_term(&mut sig, "A(y_ B(C) D)").expect("parse of A(y_ B(C) D)");
+    /// let t = parse_term(&mut sig, "B(C)").expect("parse of B(C)");
+    /// let v = parse_term(&mut sig, "x_").expect("parse of x_");
+    ///
+    /// let new_term = TRS::replace_term_helper(&term, &t, v);
+    ///
+    /// assert_eq!(new_term.display(), "A(y_ x_ D)");
+    /// # }
+    /// ```
+    pub fn replace_term_helper(term: &Term, t: &Term, v: Term) -> Term {
```

Comment:
`t` and `v` aren't particularly descriptive. Could you use more memorable variable names?